static/js/calendar/calendar-model.js

```diff
@@ -0,0 +1,81 @@
+import * as dateUtil from '../date-util.js'
+import CalendarDayModel from './calendar-dayModel.js'
+import CalendarRecordModel from './calendar-recordModel.js'
+
+export default class CalendarModel {
+  constructor(holidays, year, month, recordList) {
+    this.year = year
+    this.month = month
+    this.holidays = holidays
+    this.recordList = recordList
+
+    this.dayList = this.getDayList()
+  }
+
+  static async build() {
+    const { year, month } = dateUtil.getDateJSON()
+
+    const recordList = await CalendarRecordModel.build()
+
+    const holidays = await CalendarDayModel.getHolidayList(year, month)
```

Comment:
你應該用 `Promise.all` 來同時處理兩個不相關的非同步呼叫

```suggestion
    const [recordList, holidays] = await Promise.all([CalendarRecordModel.build(), CalendarDayModel.getHolidayList(year, month)])    
```